### PR TITLE
Add screw executor

### PR DESCRIPTION
**Pull request by AdamPettinger**
- Renamed files `src/screw_motion.cpp` -> `src/screw_axis.cpp` for a little better clarity (includes header rename)
- Added screw executor, which uses a `ScrewAxis` and an EE/moving frame to calculate how a robot needs to move to follow a screw motion. The EE/moving frame can be passed via a TF message, or can be looked up in the TF tree. The screw executor has a C++ API as well as a ROS Service API
- Added Unit and ROS tests for screw executor

You should be able to run the tests locally with just `catkin test affordance_primitives`

### CODE REVIEW

**Review by sidharthNair — COMMENTED** [review]
LGTM 👍

File: CMakeLists.txt [hunk 1]
```diff
@@ -12,12 +12,14 @@ find_package(catkin REQUIRED COMPONENTS
   affordance_primitive_msgs
   geometry_msgs
   tf2_eigen
+	tf2_ros
```
Comment (FabianEP11) [inline CMakeLists.txt]: weird indentation

File: CMakeLists.txt [hunk 2]
```diff
@@ -12,12 +12,14 @@ find_package(catkin REQUIRED COMPONENTS
...
 )
 
 set(THIS_PACKAGE_INCLUDE_DEPENDS
     affordance_primitive_msgs
     geometry_msgs
     tf2_eigen
+		tf2_ros
```
Comment (FabianEP11) [inline CMakeLists.txt]: same here

**Review by FabianEP11 — COMMENTED** [review]
Looks good for me. 
Just one comment: The test works for a 2 meters translation and 90 degree rotation over x. I'm guessing those hardcode values represent a configuration for an specific case. It would be nice to pass those values as arguments, maybe using argc/argv, in case I want to try a different configuration.

File: src/screw_execution.cpp [hunk 3]
```diff
@@ -0,0 +1,70 @@
+#include <affordance_primitives/screw_execution.hpp>
+#include <affordance_primitives/affordance_utils.hpp>
+#include <affordance_primitives/screw_axis.hpp>
+
+namespace affordance_primitives
+{
+APScrewExecutor::APScrewExecutor() : tfBuffer_(), tfListener_(tfBuffer_)
```
Comment (EAkita) [inline src/screw_execution.cpp]: I'm not sure if the first `tfBuffer_( )` is needed in this initializer list. 
`tfListener_(tfBuffer_)` works just fine.

File: src/screw_execution.cpp [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+{
+}
+
+APScrewExecutor::APScrewExecutor(ros::NodeHandle& nh, const std::string& server_name_)
+  : tfBuffer_(), tfListener_(tfBuffer_)
```
Comment (EAkita) [inline src/screw_execution.cpp]: Same here with the first `tfBuffer_( )` in the initializer list...  unless I'm missing something?

**Review by EAkita — APPROVED** [review]
Looks good man! You could address the `tf_buffer()` comment when you get a chance.

File: src/screw_execution.cpp (hunk 3 above)
Comment (AdamPettinger) [reply src/screw_execution.cpp]: It's true, I removed it. `tf2_ros::Buffer` has a default constructor which is used in both cases, which is why it behaves the same

**Comment by AdamPettinger** [comment]
> Just one comment: The test works for a 2 meters translation and 90 degree rotation over x. I'm guessing those hardcode values represent a configuration for an specific case. It would be nice to pass those values as arguments, maybe using argc/argv, in case I want to try a different configuration.

I would love to discuss more. That test case isn't a specific case I use anywhere else, it was just easy to calculate by hand. To be able to have any TF and screw axis in the test would mean redoing a lot of the math that is getting tested inside the test itself, which feels funny to me

**Comment by AdamPettinger** [comment]
> Just one comment: The test works for a 2 meters translation and 90 degree rotation over x. I'm guessing those hardcode values represent a configuration for an specific case. It would be nice to pass those values as arguments, maybe using argc/argv, in case I want to try a different configuration.

I would love to discuss more. That test case isn't a specific case I use anywhere else, it was just easy to calculate by hand. To be able to have any TF and screw axis in the test would mean redoing a lot of the math that is getting tested inside the test itself, which feels funny to me

**Pull request merged**